python/tk_nuke_writenode/handler.py

```diff
@@ -1654,8 +1655,19 @@ def __gather_render_settings(self, node, is_proxy=False):
             # check for 'channel' for backwards compatibility
             if "output" in render_template.keys or "channel" in render_template.keys:
                 output_name = node.knob(TankWriteNodeHandler.OUTPUT_KNOB_NAME).value()
-            
-        return (render_template, width, height, output_name)
+
+        # extension = node.knob('tk_file_type').value()
+
+        # decode extension from profile['file_extension']
+        # File open fires knob changed callbacks which call this even if path is cached.
+        # This is a problem if we are trying to restore settings
+        # of a node whose profile has been deleted.
+        # Setting extension to file_type in that case.
+        profile_name = node.knob('profile_name').value()
```

Comment:
I get this, but I'd like to see a more robust solution for this. Please look at how the "file_type" attribute is handled in the _set_profile() and _populate_format_settings() methods. We should handle file_extension in the same way, so that we can better handle those situations like when the profile isn't found.

Reply:
We can add file_extension as a knob to WriteTank gizmo and store it there (as you said, similar to file_type).

I still don't see why we have to define file extension here though. Why not just use file_type as extension and handle the renaming during delivery?

Reply:
Its a useful feature, but I agree that the rename could just be done at delivery...unless we want to use .jpg instead of .jpeg internally as well.

Reply:
Added file_extension as a knob. I don't know how it will affect any existing WriteTank nodes though. I'm not sure the feature is worth the modification, because I don't see any reason to restrict our internal name to only .jpg.